packages/foam-vscode/src/features/tags-tree-view/index.ts

```diff
@@ -82,6 +87,11 @@ export class TagsProvider implements vscode.TreeDataProvider<TagTreeItem> {
 
       const references: TagTreeItem[] = element.notes
         .map(({ uri }) => this.foam.workspace.get(uri))
+        .filter(note =>
+          // foreach nested tag item check if the exact tag is present as a tag on the note. If so,
+          // it belongs to a nested tag item and should not be displayed as a reference underneath the current element
+          isNone(nestedTagItems.find(item => note.tags.has(item.tag)))
```

Comment:
I've been looking why the problem exists anyway. Turns out I've added all the complexity because I didn't assign the right tag to the parent item. With the right tag on that element, it is easier to filter for matches. Should be more readable now and function as it should.